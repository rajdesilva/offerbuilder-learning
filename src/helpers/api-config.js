import { init as initApm } from "@elastic/apm-rum";

export const baseURL = {
  API_URL_BASE: process.env.REACT_APP_BASE_API_URL,
  LOGIN_URL: process.env.REACT_APP_LOGIN_URL,
  LOGOUT_URL: process.env.REACT_APP_LOGOUT_URL,
  PROFILE_URL: process.env.REACT_APP_USER_PROFILE_URL,
  HISTORY_URL: process.env.REACT_APP_USER_HISTORY_URL,
};

export const apm = initApm({
  // Set required service name (allowed characters: a-z, A-Z, 0-9, -, _, and space)
  serviceName: process.env.REACT_APP_APM_SERVICE_NAME,
  serverUrl: process.env.REACT_APP_APM_SERVICE_URL,

  // Set service version (required for sourcemap feature)
  serviceVersion: process.env.REACT_APP_APM_SERVICE_VERSION,
});