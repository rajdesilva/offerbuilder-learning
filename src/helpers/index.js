export * from './api-config';
export * from './history';
