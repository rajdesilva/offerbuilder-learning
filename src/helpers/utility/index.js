export { createSearchQueryURL } from './createSearchQueryURL';
export { isParentOrChildrenPropertyAdded } from './isParentOrChildrenPropertyAdded';
export { isPropertySelected } from './isPropertySelected';
export { mapBrands } from './mapBrands';
export { mapOnlySuppliers } from './mapOnlySuppliers';
export { parseFormDescriptionsAndImagesForOffer } from './parseFormDescriptionsAndImagesForOffer';
export { removePropertyForSelectedChannels } from './removePropertyForSelectedChannels';
export { removePropertyForSelectedSuppliers } from './removePropertyForSelectedSuppliers';
export { getParsedProperty } from './getParsedProperty';
export { getFormattedValue } from './getFormattedValue';
export { getValue } from './getValue';
export { parsePropertyMarketingInfoToFormData } from './parsePropertyMarketingInfoToFormData';
export { createOfferRequestObject } from './createOfferRequestObject';
export { getParsedOffers } from './getParsedOffers';
export { getListOfPropertiesForSupplier } from './getListOfPropertiesForSupplier';
export { getListOfPropertiesForChannels } from './getListOfPropertiesForChannels';
export { getInitializedState } from './getInitializedState';
export { getParsedNewOfferSettingsObject } from './getParsedNewOfferSettingsObject';
export { getListOfPropertiesForStorefront } from './getListOfPropertiesForStorefront';
export { checkIfMultipleChannelPresentForTarget } from './checkIfMultipleChannelPresentForTarget';
export { checkIfMultipleSupplierPresentForTarget } from './checkIfMultipleSupplierPresentForTarget';
export { checkIfMultipleStorefrontPresentForTarget } from './checkIfMultipleStorefrontPresentForTarget';
export { removePropertyForSelectedStorefront } from './removePropertyForSelectedStorefront';
export { checkIfMultipleBrandPresentForTarget } from './checkIfMultipleBrandPresentForTarget';
export { getListOfPropertiesForBrand } from './getListOfPropertiesForBrand';
export { getSearchStateFromRedux } from './getSearchStateFromRedux';
export { getObjectStringify } from './getObjectStringify';
export { parseOfferMarketingFromOffer } from './parseOfferMarketingFromOffer';
export { parseOfferMarketingLanguagesFromOffer } from './parseOfferMarketingLanguagesFromOffer';
export { getFilterData } from './getFilterData';
export { getSearchAndFilterQueryURL } from './getSearchAndFilterQueryURL';
export { getSearchQueryURLForEditAddProperties } from "./getSearchQueryURLForEditAddProperties";
export { isSavedOfferProperty } from './isSavedOfferProperty';
export { getParsedPropertiesForDeeplinkSettings } from "./getParsedPropertiesForDeeplinkSettings";
export { resetApplicationReduxAndStorage } from "./resetApplicationReduxAndStorage";
export { updateSettingsBrandWithSearchedValues } from "./updateSettingsBrandWithSearchedValues";
export { checkIfUserHasRole } from "./checkIfUserHasRole";
export { getUserOfferBuilderRole } from "./getUserOfferBuilderRole";
export { checkIfNewPropertiesAdded } from "./checkIfNewPropertiesAdded";
export { getAllBrandsWithoutKeysFromTempBrands } from "./getAllBrandsWithoutKeysFromTempBrands";
export { checkIfItemAlreadyPresentInList } from "./checkIfItemAlreadyPresentInList";
export { checkIfTravellingDateValidAsPerBookingDate } from "./checkIfTravellingDateValidAsPerBookingDate";
export { checkIfPropertiesHaveMultipleLocations } from "./checkIfPropertiesHaveMultipleLocations";
export { displayMargin } from "./displayMargin";
export { getFilterCount } from "./getFilterCount";
export { labelForBrands } from "./labelForBrands";
export { getDistanceFromLatLonInKm } from "./getDistanceFromLatLonInKm";
export { getTrustYouIdListFromProperties } from "./getTrustYouIdListFromProperties";
export { getSearchParamsToFetchMarketPrice } from "./getSearchParamsToFetchMarketPrice";
export { getMarketPriceForPropertyFromList } from "./getMarketPriceForPropertyFromList";
export { getImageFilterCount } from "./getImageFilterCount";
export { getSearchAndFilterImagesQueryUrl } from "./getSearchAndFilterImagesQueryUrl";
export { getImageName } from "./getImageName";
export { updateReduxWithOfferData } from "./updateReduxWithOfferData";
export { getOfferTypeFromPropertiesSelected } from "./getOfferTypeFromPropertiesSelected";