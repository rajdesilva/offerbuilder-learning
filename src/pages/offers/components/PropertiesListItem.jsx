import React, { Fragment, useState } from "react";
import { List, Button } from "antd";
import { FormattedMessage } from "react-intl";
import PropTypes from "prop-types";
import styles from "./css/PropertiesListItem.module.less";

const getPropertiesForList = (properties) => {
  return properties.length % 2 === 0
    ? properties
    : [...properties, { key: "collapse" }];
};

function PropertiesListItem({ row, dataTestId }) {
  const properties = getPropertiesForList(row.properties);

  const [showMore, setShowMore] = useState(false);
  return (
    <List
      grid={{
        gutter: 16,
        xs: 2,
        sm: 2,
        md: 2,
        lg: 2,
        xl: 2,
        xxl: 2,
      }}
      data-testid={dataTestId}
      dataSource={properties}
      renderItem={(item, index) => {
        return index === 7 &&
          row.properties.length > 7 &&
          showMore === false ? (
          <List.Item
            className={styles.listItem}
            onClick={() => {
              setShowMore(!showMore);
            }}
          >
            <Button
              type="link"
              className={styles.showMoreBtn}
              data-testid={dataTestId + "-show-more"}
            >
              <Fragment>
                <FormattedMessage id="nemo.and" /> {row.properties.length - 7}{" "}
                <FormattedMessage id="nemo.more" />
              </Fragment>
            </Button>
          </List.Item>
        ) : index > 7 && showMore === false ? (
          <Fragment />
        ) : index === row.properties.length && showMore === true ? (
          <List.Item className={styles.listItem}>
            <Button
              type="link"
              className={styles.showMoreBtn}
              onClick={() => setShowMore(!showMore)}
              data-testid={dataTestId + "-collapse-all"}
            >
              <FormattedMessage id="nemo.collapseAll" />
            </Button>
          </List.Item>
        ) : (
          <List.Item className={styles.listItem}>
            <Fragment>
              <div className={styles.propertyName}>{item.name}</div>
              {index === row.properties.length - 1 &&
              showMore === true &&
              row.properties.length % 2 === 0 ? (
                <Button
                  type="link"
                  className={styles.showMoreBtn}
                  onClick={() => setShowMore(!showMore)}
                  data-testid={dataTestId + "-collapse-all"}
                >
                  <FormattedMessage id="nemo.collapseAll" />
                </Button>
              ) : (
                <Fragment />
              )}
            </Fragment>
          </List.Item>
        );
      }}
    />
  );
}

PropertiesListItem.propTypes = {
  row: PropTypes.object,
  dataTestId: PropTypes.string,
};
export default PropertiesListItem;
