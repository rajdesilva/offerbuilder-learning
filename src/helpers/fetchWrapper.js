import { message } from "antd";
import { createIntl, createIntlCache } from "@formatjs/intl";
import * as languages from "../langs/";
import { baseURL } from "./api-config";
import { history } from "./history";

// This is optional but highly recommended
// since it prevents memory leak
const cache = createIntlCache();

// Create the `intl` object
const intl = createIntl(
  {
    // Locale of the application
    // TODO:
    locale: "en",
    // Locale of the fallback defaultMessage
    defaultLocale: "en",
    messages: languages["en"],
  },
  cache
);
const defaultRequest = {
  credentials: "include",
  headers: {
    "Content-Type": "application/json",
  },
};
export const fetchWrapper = (url, payload) => {
  payload = {
    ...defaultRequest,
    ...payload,
  };
  return fetch(url, payload).then(
    (response) => {
      // Check that the response is valid and reject an error
      // response to prevent subsequent attempt to parse json
      if (!response.ok) {
        if (response.status === 500) {
          message.error(
            intl.formatMessage({
              id: "nemo.error500msg",
            })
          );
        } else if (response.status === 503) {
          message.error(
            intl.formatMessage({
              id: "nemo.error503msg",
            })
          );
        } else if (response.status === 504) {
          message.error(
            intl.formatMessage({
              id: "nemo.error504msg",
            })
          );
        } else {
          message.error(
            intl.formatMessage({
              id: "nemo.somethingWentWrongPleaseTryAgain",
            })
          );
        }
        return;
      }
      if (response.status === 403) {
        history.push("/access-denied");
      }
      return response.json ? response.json() : response;
    },
    (error) => {
      /**
       * user url, payload and error here for sending to the logger.
       */
      if (error instanceof TypeError) {
        window.location.href = baseURL.API_URL_BASE + "redirect";
      }
      console.error("fetch error : ", error);
    }
  );
};
