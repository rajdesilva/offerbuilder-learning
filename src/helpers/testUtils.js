import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render as rtlRender } from "@testing-library/react";
import { getValue } from "./utility";
import rootReducers from "../redux/rootReducers";
import * as languages from "../langs";
import { IntlProvider } from "react-intl";
const initialReducerState = {};
window.getValue = getValue;

window.google = {
  maps: {
    Marker: class {},
    Map: class {},
    LatLngBounds: class {},
    places: {
      Autocomplete: class {},
      AutocompleteService: class {},
      PlacesServiceStatus: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
      PlacesAutocomplete: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
    },

    MarkerClusterer: class {},
    Geocoder: class {},
  },
};
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

function render(
  ui,
  {
    initialState = initialReducerState,
    store = createStore(rootReducers, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <IntlProvider locale="en" messages={languages["en"]}>
          {children}
        </IntlProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { render };
