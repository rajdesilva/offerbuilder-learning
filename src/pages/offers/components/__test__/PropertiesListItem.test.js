import React from "react";
import { render, fireEvent } from "../../../../helpers/testUtils";
import PropertiesListItem from "../PropertiesListItem";

const properties = [
  {
    internalId: 4200,
    propertyCode: "DEMO_JEEVAN",
    name: "AJSG Hotel ",
    channel: "NEMO_CH_03",
    supplier: "ntp",
    city: "Vienna",
    rating: 0,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: false,
    lowestMargin: 5,
    highestMargin: 5,
    marketPrice: null,
    images: [],
    latitude: 48.19912,
    longitude: 16.349869,
    sequence: 1,
    hotelName: "AJSG Hotel ",
    key: 0,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [
      {
        lang: "DE",
        shortDescription: null,
        description: "Duch lang ggjjgjgjgjgj",
        name: "AJSG Hotel (DE)",
      },
      {
        lang: "EN",
        shortDescription: null,
        description:
          "Our luxurious Apartment hotel in country style is situated directly in the untouched nature of a side valley of the Vinschgau. At our house you can relax 24 hours a day: Start your day with a hearty breakfast and afterwards enjoy the beautiful nature and the various sport activities we have to offer. Relax in our alpine sauna world or our Panorama swimming pool. In the evening you can choose between three different restaurants in which we serve regional and international specialties.",
        name: "AJSG Hotel ",
      },
    ],
  },
  {
    internalId: 4201,
    propertyCode: "XS_NM_0001",
    name: "Bishop's Lodge",
    channel: "DEMO_INDIA_CH",
    supplier: "ntp",
    city: "Santa Fe",
    rating: 5,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: true,
    lowestMargin: 15,
    highestMargin: 15,
    marketPrice: null,
    images: [],
    latitude: 35.715103,
    longitude: -105.91955,
    sequence: 2,
    hotelName: "Bishop's Lodge",
    key: 1,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [
      {
        lang: "DE",
        shortDescription: null,
        description:
          "Unser Resort ist ein Reiseziel wie kein anderes und bietet gehobene Ranchunterkünfte und endlose Erholungsmöglichkeiten für vier Jahreszeiten. Genießen Sie klare, sonnenverwöhnte Tage und erkunden Sie unsere 450 Hektar großen malerischen Wanderwege vom Reiten, Wandern und Mountainbiken.",
        name: "Bishop's Lodge",
      },
      {
        lang: "EN",
        shortDescription: null,
        description:
          "A destination unlike any other, our resort offers upscale ranch lodging and endless four-season recreation. Enjoy clear, sun-drenched days exploring our 450 acres of scenic trails from horseback riding, hiking and mountain biking.",
        name: "Bishop's Lodge",
      },
    ],
  },
  {
    internalId: 4202,
    propertyCode: "DEMO_TEST001",
    name: "DEMO PRADEEP3",
    channel: "DEMO_OFFERBUILDER",
    supplier: "ntp",
    city: "Vienna",
    rating: 0,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: false,
    lowestMargin: null,
    highestMargin: null,
    marketPrice: null,
    images: [],
    latitude: 48.19912,
    longitude: 16.349869,
    sequence: 3,
    hotelName: "DEMO PRADEEP3",
    key: 2,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [],
  },
  {
    internalId: 4203,
    propertyCode: "DEMO_SUNIL",
    name: "DEMO_SUNIL",
    channel: "DEMO_OFFERBUILDER",
    supplier: "ntp",
    city: "Vienna",
    rating: 0,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: false,
    lowestMargin: null,
    highestMargin: null,
    marketPrice: null,
    images: [],
    latitude: 48.19912,
    longitude: 16.349869,
    sequence: 4,
    hotelName: "DEMO_SUNIL",
    key: 3,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [
      {
        lang: "DE",
        shortDescription: null,
        description: "Marketing long description in Deutsh",
        name: "DEMO_SUNIL",
      },
      {
        lang: "EN",
        shortDescription: null,
        description: "Marketing long description in English",
        name: "DEMO_SUNIL",
      },
    ],
  },
  {
    internalId: 4204,
    propertyCode: "DEMO_BANJARE",
    name: "English B&B Hotel",
    channel: "DEMO_OFFERBUILDER",
    supplier: "ntp",
    city: "Vienna",
    rating: 4,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: false,
    lowestMargin: null,
    highestMargin: null,
    marketPrice: null,
    images: [],
    latitude: 48.2082,
    longitude: 16.3738,
    sequence: 5,
    hotelName: "English B&B Hotel",
    key: 4,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [],
  },
  {
    internalId: 4205,
    propertyCode: "DEMO_GHULE",
    name: "GHULE ROCK MANSION",
    channel: "DEMO_OFFERBUILDER",
    supplier: "ntp",
    city: "Vienna",
    rating: 4,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: false,
    lowestMargin: 20,
    highestMargin: 20,
    marketPrice: null,
    images: [],
    latitude: 48.19912,
    longitude: 16.349869,
    sequence: 6,
    hotelName: "GHULE ROCK MANSION",
    key: 5,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [],
  },
  {
    internalId: 4206,
    propertyCode: "DEMO_HEERO",
    name: "HEERO Merriott STAR",
    channel: "DEMO_OFFERBUILDER",
    supplier: "ntp",
    city: "Vienna",
    rating: 4,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: false,
    lowestMargin: 8,
    highestMargin: 8,
    marketPrice: null,
    images: [],
    latitude: 48.19912,
    longitude: 16.349869,
    sequence: 7,
    hotelName: "HEERO Merriott STAR",
    key: 6,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [],
  },
  {
    internalId: 4207,
    propertyCode: "DEMO_PKM_002",
    name: "Hotel Paradise PKM",
    channel: "NEMO_CH_01",
    supplier: "ntp",
    city: "Vienna",
    rating: 5,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: false,
    lowestMargin: 0,
    highestMargin: 0,
    marketPrice: null,
    images: [],
    latitude: 48.19912,
    longitude: 16.349869,
    sequence: 8,
    hotelName: "Hotel Paradise PKM",
    key: 7,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [],
  },
  {
    internalId: 4208,
    propertyCode: "H000892",
    name: "Hotel Pradeep",
    channel: "KOGNITIV_QA",
    supplier: "ntp",
    city: "Pune",
    rating: 5,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: true,
    lowestMargin: 23,
    highestMargin: 23,
    marketPrice: null,
    images: [],
    latitude: 18.610317,
    longitude: 73.78897,
    sequence: 9,
    hotelName: "Hotel Pradeep",
    key: 8,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [],
  },
  {
    internalId: 4209,
    propertyCode: "DEMO_KM",
    name: "KM Royal Palace",
    channel: "DEMO_OFFERBUILDER",
    supplier: "ntp",
    city: "Vienna",
    rating: 5,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: false,
    lowestMargin: null,
    highestMargin: null,
    marketPrice: null,
    images: [],
    latitude: 48.19912,
    longitude: 16.349869,
    sequence: 10,
    hotelName: "KM Royal Palace",
    key: 9,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [],
  },
  {
    internalId: 4210,
    propertyCode: "DEMO_PRATIK",
    name: "NICE RESORT PRATIK",
    channel: "NEMO_CH_01",
    supplier: "ntp",
    city: "Vienna",
    rating: 4.5,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: false,
    lowestMargin: 0,
    highestMargin: 0,
    marketPrice: null,
    images: [],
    latitude: 48.19912,
    longitude: 16.349869,
    sequence: 11,
    hotelName: "NICE RESORT PRATIK",
    key: 10,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [],
  },
  {
    internalId: 4211,
    propertyCode: "DEMO_PRADEEP",
    name: "PRADEEP PARTY MANSION",
    channel: "KOGNITIV_QA",
    supplier: "ntp",
    city: "Vienna",
    rating: 0,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: false,
    lowestMargin: null,
    highestMargin: null,
    marketPrice: null,
    images: [],
    latitude: 48.19912,
    longitude: 16.349869,
    sequence: 12,
    hotelName: "PRADEEP PARTY MANSION",
    key: 11,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [],
  },
  {
    internalId: 4212,
    propertyCode: "DEMO_PUNJABI",
    name: "PUNJABI COOL HOTEL",
    channel: "DEMO_OFFERBUILDER",
    supplier: "ntp",
    city: "Vienna",
    rating: 0,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: false,
    lowestMargin: null,
    highestMargin: null,
    marketPrice: null,
    images: [],
    latitude: 48.19912,
    longitude: 16.349869,
    sequence: 13,
    hotelName: "PUNJABI COOL HOTEL",
    key: 12,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [],
  },
  {
    internalId: 4213,
    propertyCode: "DEMO_RAJ",
    name: "RAJ Rock Party House",
    channel: "DEMO_OFFERBUILDER",
    supplier: "ntp",
    city: "Vienna",
    rating: 0,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: false,
    lowestMargin: null,
    highestMargin: null,
    marketPrice: null,
    images: [],
    latitude: 48.19912,
    longitude: 16.349869,
    sequence: 14,
    hotelName: "RAJ Rock Party House",
    key: 13,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [],
  },
  {
    internalId: 4214,
    propertyCode: "DEMO_PR_1",
    name: "Royal Suite PRA_RV",
    channel: "DEMO_INDIA_CH",
    supplier: "ntp",
    city: "Bangalore",
    rating: 5,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: false,
    lowestMargin: null,
    highestMargin: null,
    marketPrice: null,
    images: [],
    latitude: 12.970991,
    longitude: 77.6069,
    sequence: 15,
    hotelName: "Royal Suite PRA_RV",
    key: 14,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [],
  },
  {
    internalId: 4215,
    propertyCode: "XS_WY_0001",
    name: "Santa Fe Sage Inn",
    channel: "DEMO_INDIA_CH",
    supplier: "ntp",
    city: "Santa Fe",
    rating: 3,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: false,
    lowestMargin: 0,
    highestMargin: 15,
    marketPrice: null,
    images: [],
    latitude: 35.67994,
    longitude: -105.94874,
    sequence: 16,
    hotelName: "Santa Fe Sage Inn",
    key: 15,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [],
  },
  {
    internalId: 4216,
    propertyCode: "DEMO_SWAP_01",
    name: "Swap Paradise Star1",
    channel: "NEMO_CH_02",
    supplier: "ntp",
    city: "Vienna",
    rating: 4,
    ratingProvider: "",
    remainingCapitalPool: 0,
    lcn: false,
    lowestMargin: 5,
    highestMargin: 5,
    marketPrice: null,
    images: [],
    latitude: 48.19912,
    longitude: 16.349869,
    sequence: 17,
    hotelName: "Swap Paradise Star1",
    key: 16,
    isSavedProperty: true,
    marketingImages: [],
    descriptions: [],
  },
];
describe("PropertiesListItem component test", () => {
  test("create PropertiesListItem component snapshot", () => {
    const { container } = render(
      <PropertiesListItem
        row={{ properties }}
        dataTestId="A1-active-offer-item-properties-list"
      />,
      {}
    );

    expect(container.firstChild).toMatchSnapshot();
  });
  test(" PropertiesListItem - no properties message", () => {
    const { getByText } = render(
      <PropertiesListItem
        row={{ properties: [] }}
        dataTestId="A1-active-offer-item-properties-list"
      />,
      {}
    );

    expect(getByText(/No Data/i)).toBeInTheDocument();
  });
  test("PropertiesListItem component - active offers tab -  show more button is visible and click on it and it will show collapse all button", async () => {
    const { getByTestId } = render(
      <PropertiesListItem
        row={{ properties }}
        dataTestId="A1-active-offer-item-properties-list"
      />,
      {}
    );
    // check for show more button in the DOM
    expect(
      getByTestId(
        "A1-active-offer-item-properties-list-show-more"
      ).textContent.trim()
    ).toEqual("And 10 more");
    //click on show more button
    await fireEvent.click(
      getByTestId("A1-active-offer-item-properties-list-show-more")
    );
    // after clicking on show more button collapse all button should be in the DOM
    expect(
      getByTestId("A1-active-offer-item-properties-list-collapse-all")
    ).toBeInTheDocument();
    //click on collapse-all button
    await fireEvent.click(
      getByTestId("A1-active-offer-item-properties-list-collapse-all")
    );
    // after clicking on collapse-all button we should have show more button in the DOM
    expect(
      getByTestId(
        "A1-active-offer-item-properties-list-show-more"
      ).textContent.trim()
    ).toEqual("And 10 more");
  });
});
