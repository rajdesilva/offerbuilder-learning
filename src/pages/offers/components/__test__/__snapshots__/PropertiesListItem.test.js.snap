// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertiesListItem component test create PropertiesListItem component snapshot 1`] = `
<div
  class="ant-list ant-list-split ant-list-grid"
  data-testid="A1-active-offer-item-properties-list"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="ant-row"
        style="margin-left: -8px; margin-right: -8px;"
      >
        <div>
          <div
            class="ant-col"
            style="padding-left: 8px; padding-right: 8px; flex: 1 1 auto;"
          >
            <div
              class="ant-list-item"
            >
              <div>
                AJSG Hotel 
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="ant-col"
            style="padding-left: 8px; padding-right: 8px; flex: 1 1 auto;"
          >
            <div
              class="ant-list-item"
            >
              <div>
                Bishop's Lodge
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="ant-col"
            style="padding-left: 8px; padding-right: 8px; flex: 1 1 auto;"
          >
            <div
              class="ant-list-item"
            >
              <div>
                DEMO PRADEEP3
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="ant-col"
            style="padding-left: 8px; padding-right: 8px; flex: 1 1 auto;"
          >
            <div
              class="ant-list-item"
            >
              <div>
                DEMO_SUNIL
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="ant-col"
            style="padding-left: 8px; padding-right: 8px; flex: 1 1 auto;"
          >
            <div
              class="ant-list-item"
            >
              <div>
                English B&B Hotel
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="ant-col"
            style="padding-left: 8px; padding-right: 8px; flex: 1 1 auto;"
          >
            <div
              class="ant-list-item"
            >
              <div>
                GHULE ROCK MANSION
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="ant-col"
            style="padding-left: 8px; padding-right: 8px; flex: 1 1 auto;"
          >
            <div
              class="ant-list-item"
            >
              <div>
                HEERO Merriott STAR
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="ant-col"
            style="padding-left: 8px; padding-right: 8px; flex: 1 1 auto;"
          >
            <div
              class="ant-list-item"
            >
              <button
                class="ant-btn ant-btn-link"
                data-testid="A1-active-offer-item-properties-list-show-more"
                type="button"
              >
                And
                 
                10
                 
                more
              </button>
            </div>
          </div>
        </div>
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;
